docs/concepts/time-and-windows-in-ksql-queries.rst

```diff
@@ -0,0 +1,305 @@
+============
+
+There are three ways to define time windows in KSQL.
+
+.. image:: ../img/ksql-window-aggregation.png
+   :alt: Diagram showing three types of time windows in KSQL streams: tumbling, hopping, and session
+
+Tumbling Window
+---------------
+
+All time windows are the same size and adjacent to each other, which
+means that whenever a window ends, the next window starts.
+
+For example, if you want to compute the the five highest-value orders
+per zip code per hour in an ``orders`` stream, you might run a query like this:
+
+.. code:: sql
+
+    SELECT orderzip_code, TOPK(order_total, 5) FROM orders
+      WINDOW TUMBLING (SIZE 1 HOUR) GROUP BY order_zipcode;
+
+Here's another example: to detect potential credit card fraud in an
+``authorization_attempts`` stream, you might run a query for the number of
+authorization attempts on a particular card that's greater than three, during
+a time interval of five seconds.
+
+.. code:: sql
+
+    SELECT card_number, count(*) FROM authorization_attempts
+      WINDOW TUMBLING (SIZE 5 SECONDS)
+      GROUP BY card_number HAVING count(*) > 3;
+
+The tumbling window's start time is inclusive, but the end time is exclusive.
+This is important for non-overlapping windows, in which each record must be
+contained in exactly one window.
+
+Hopping Window
+--------------
+
+All time windows are the same size, but they might overlap, depending on the
+length of time specified in the ADVANCE BY property.
+
+For example, if you want to count the pageviews for only ``Region_6`` by female
+users for a hopping window of 30 seconds that advances by 10 seconds, you might
+run a query like this:
+
+.. code:: sql
+
+    SELECT regionid, count(*) FROM pageviews 
+      WINDOW HOPPING (SIZE 30 SECONDS, ADVANCE BY 10 SECONDS)
+      WHERE UCASE(gender)='FEMALE' AND LCASE (regionid) LIKE '%_6'
+      GROUP BY regionid;
+
+The hopping window's start time is inclusive, but the end time is exclusive.
+This is important for non-overlapping windows, in which each record must be
+contained in exactly one window.
+
+Session Window
+--------------
+
+A new window starts if the last event that arrived is further back in time
+than a specified session timeout time.
+
+For example, to count the number of pageviews per region for session windows
+with a session inactivity gap of 60 seconds, you might run the following query,
+which "sessionizes" the input data and performs the counting/aggregation step
+per region:
+
+.. code:: sql
+
+    SELECT regionid, count(*) FROM pageviews
+      WINDOW SESSION (60 SECONDS)
+      GROUP BY regionid;
+
+The start and end times for a session window are both inclusive, in contrast to
+time windows. There is always a record in the session window with both the start
+and end timestamps, because the timestamp of the first and last record in the
+window define the session window's start and end time.
+
+Windowed Joins
+--------------
+
+KSQL supports using windows in JOIN queries. 

```

Comment:
Nit: "KSQL supports using windows in JOIN queries via a `WITHIN` clause."

Reply:
Also:

- Support for WITHIN: Any restrictions for WITHIN, or can we use it for any join we support?
- A stream-stream must always be windowed, right?  We should also call this out in the new docs on Joins.
- Do we want to cover windowed joins under joins, or here?  Or in both places?